azurerm/internal/services/containers/kubernetes_cluster_pod_identity_resource.go

```diff
@@ -0,0 +1,355 @@
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.ManagedClusterName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Managed Kubernetes Cluster %q was not found in Resource Group %q - removing from state!", id.ManagedClusterName, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	if resp.ManagedClusterProperties == nil ||
+		resp.ManagedClusterProperties.PodIdentityProfile == nil ||
+		resp.ManagedClusterProperties.PodIdentityProfile.Enabled == nil ||
+		!*resp.ManagedClusterProperties.PodIdentityProfile.Enabled {
+		log.Printf("[DEBUG] Pod Identity Managed Kubernetes Cluster %q was not found in Resource Group %q - removing from state!", id.ManagedClusterName, id.ResourceGroup)
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("cluster_id", resp.ID)
+	if err := d.Set("pod_identity", flattenKubernetesPodIdentities(resp.ManagedClusterProperties.PodIdentityProfile.UserAssignedIdentities)); err != nil {
+		return fmt.Errorf("setting `pod_identity`: %+v", err)
+	}
+
+	if err := d.Set("exception", flattenKubernetesPodIdentityException(resp.ManagedClusterProperties.PodIdentityProfile.UserAssignedIdentityExceptions)); err != nil {
+		return fmt.Errorf("setting `exception`: %+v", err)
+	}
+
+	return nil
+}
+
+func resourceKubernetesClusterPodIdentityDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Containers.KubernetesClustersClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ClusterID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	existing, err := client.Get(ctx, id.ResourceGroup, id.ManagedClusterName)
+	if err != nil {
+		return fmt.Errorf("checking for presence of existing %s: %s", id, err)
+	}
+
+	if existing.ManagedClusterProperties == nil {
+		return fmt.Errorf("`ManagedClusterProperties` is nil for %s: %s", id, err)
+	}
+
+	existing.ManagedClusterProperties.PodIdentityProfile = &containerservice.ManagedClusterPodIdentityProfile{
+		Enabled: utils.Bool(false),
+	}
+
+	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.ManagedClusterName, existing)
+	if err != nil {
+		return fmt.Errorf("deleting Pod Identity for %s: %+v", id, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for deletion of Pod Identity for %s: %+v", id, err)
+	}
+
+	return nil
+}
+
+func expandKubernetesPodIdentities(ctx context.Context, userAssignedIdentitiesClient *msi.UserAssignedIdentitiesClient, input []interface{}) (*[]containerservice.ManagedClusterPodIdentity, error) {
+	if len(input) == 0 {
+		return nil, nil
+	}
+	result := make([]containerservice.ManagedClusterPodIdentity, 0)
+	for _, raw := range input {
+		v := raw.(map[string]interface{})
+
+		name := v["name"].(string)
+		namespace := v["namespace"].(string)
+		if name == "" || namespace == "" {
```

Comment:
It seems there is a bug with TypeSet when using current terraform-plugin-sdk. When remove the property (type set) in update operation, we could still get it by `d.Get()`, the arr size is 1 but not 0, with all elements empty.

so check empty here is to prevent this bug

Reply:
Persumably this only exists when your containing block has a `Map` (see: https://github.com/hashicorp/terraform-plugin-sdk/issues/588). Therefore, it is safe to remove these lines here.